### ISSUE TITLE
[Coverage] line immediately following ```goto``` incorrectly marked as covered

### ISSUE THREAD

**Issue by erkaii** [llvm/llvm-project#157959]
How to reproduce it (Compiler explorer [link](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:9,positionColumn:2,positionLineNumber:9,selectionStartColumn:2,selectionStartLineNumber:9,startColumn:2,startLineNumber:9),source:'int+main(void)+%7B%0A++++if+(0)+%7B%0A++++++++goto+L%3B+//+goto,+break+etc+inside+%22if%22%0A++++%7D%0A++++goto+L%3B+//+goto,+break+etc+immediately+after+%22if%22%0A++++int+g%3B%0AL:++%0A++++return+0%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'0',intel:'1',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-fprofile-instr-generate+-fcoverage-mapping',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+clang+(trunk)+(Editor+%231)',t:'0')),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:tool,i:(args:'',argsPanelShown:'0',compilerName:'x86-64+clang+(trunk)',editorid:1,fontScale:14,fontUsePx:'0',j:1,monacoEditorHasBeenAutoOpened:'1',monacoEditorOpen:'1',monacoStdin:'1',stdin:'',stdinPanelShown:'1',toolId:llvm-covtrunk,treeid:0,wrap:'1'),l:'5',n:'0',o:'llvm-cov+(clang-only)+x86-64+clang+(trunk)+(Editor+%231,+Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)):

Hit the bug when measuring coverage for code like:
1. [procps-4.0.2/src/ps/global.c:370](https://sources.debian.org/src/procps/2%3A4.0.2-3/src/ps/global.c#L370)
2. [lzo2-2.10/src/lzo1f_1.c:137](https://sources.debian.org/src/lzo2/2.10-2/src/lzo1f_1.c#L137)

```bash
cat > test.c << EOF
int main(void) {
    if (0) {
        goto L;
    }
    goto L;
    int g;
L:
    return 0;
}
EOF
clang -fprofile-instr-generate -fcoverage-mapping test.c -o test
./test
llvm-profdata merge default.profraw -o default.profdata
llvm-cov show -instr-profile default.profdata test
```

Coverage report where line 6 is incorrectly marked as covered:

```
    1|      1|int main(void) {
    2|      1|    if (0) {
    3|      0|        goto L;
    4|      0|    }
    5|      1|    goto L;
    6|      1|    int g;
    7|      1|L:
    8|      1|    return 0;
    9|      0|}
```

The necessary conditions to trigger this bug:
1. An ```if``` statement containing a control-flow change statement.
2. A ```goto``` statement immediately following the ```if``` statement.

Tests done with
```
clang --version
Ubuntu clang version 20.1.8 (++20250708082409+6fb913d3e2ec-1~exp1~20250708202428.132)
Target: x86_64-pc-linux-gnu
Thread model: posix
```
